Classes/Controllers/PBHistorySearchMode.h

```diff
@@ -0,0 +1,25 @@
+//
+//  PBHistorySearchMode.h
+//  GitX
+//
+//  Created by Sven-S. Porst on 2016-12-14
+//
+
+#import <Foundation/NSObjCRuntime.h>
+
+#ifndef PBHistorySearchMode_h
+#define PBHistorySearchMode_h
+
+typedef enum historySearchModes {
+	kGitXBasicSeachMode = 1,
+	kGitXPickaxeSearchMode,
+	kGitXRegexSearchMode,
+	kGitXPathSearchMode,
+	kGitXMaxSearchMode    // always keep this item last
+} PBHistorySearchMode;
+
+
+PBHistorySearchMode searchModeForInteger(NSInteger modeInteger);
```

Comment:
Good point about the naming. ✔︎

I thought it would be weird to move the enum to the Controller.h because it’s also use in the CLI gitx.c file which probably shouldn’t need to know about the controller. That‘s why I chose the separate file.